common/deployment/test_scenarios.sh

```diff
@@ -36,110 +36,121 @@ declare -A test_scenario1=(
     [modes]="FULL QUICK PUBLISH"
 )
 declare -A test_scenario2=(
-    [name]="02-oidc_password_grant"
+    [name]="02-oidc_auth_code_redirect_without_consent"
+    [display_name]="OIDC Auth Code Grant Redirect Without Consent"
+    [description]="Obtain an access token and an id token using the OAuth 2.0 authorization code redirect without consent."
+    [jmx]="oidc/OIDC_AuthCode_Redirect_WithoutConsent.jmx"
+    [tenantMode]=false
+    [skip]=false
+    [modes]="FULL QUICK PUBLISH"
+)
+declare -A test_scenario3=(
+    [name]="03-oidc_auth_code_redirect_without_consent_retrieve_user_attributes"
+    [display_name]="OIDC Auth Code Grant Redirect Without Consent Retrieve User Attributes"
+    [description]="Obtain an access token and an id token using the OAuth 2.0 authorization code grant type."
+    [jmx]="oidc/OIDC_AuthCode_Redirect_WithoutConsent_Retrieve_User_Attributes.jmx"
+    [tenantMode]=false
+    [skip]=false
+    [modes]="FULL QUICK PUBLISH"
+)
+declare -A test_scenario4=(
+    [name]="04-oidc_auth_code_redirect_without_consent_retrieve_user_attributes_groups_and_roles"
+    [display_name]="OIDC Auth Code Grant Redirect Without Consent Retrieve User Attributes Groups and Roles"
+    [description]="Obtain an access token and an id token using the OAuth 2.0 authorization code grant type."
+    [jmx]="oidc/OIDC_AuthCode_Redirect_WithoutConsent_Retrieve_User_Attributes_Groups_And_Roles.jmx"
+    [tenantMode]=false
+    [skip]=false
+    [modes]="FULL QUICK PUBLISH OIDC_AUTH_CODE_REDIRECT_WITHOUT_CONSENT_UA_GROUPS_ROLES_FLOW"
+)
+declare -A test_scenario5=(
+    [name]="05-oidc_password_grant"
     [display_name]="OIDC Password Grant Type"
     [description]="Obtain an access token and an id token using the OAuth 2.0 password grant type."
     [jmx]="oidc/OIDC_Password_Grant.jmx"
     [tenantMode]=false
     [skip]=false
     [modes]="FULL QUICK PUBLISH"
 )
-declare -A test_scenario3=(
-    [name]="03-saml2_sso_redirect_binding"
+declare -A test_scenario6=(
+    [name]="06-saml2_sso_redirect_binding"
     [display_name]="SAML2 SSO Redirect Binding"
     [description]="Obtain a SAML 2 assertion response using redirect binding."
     [jmx]="saml/SAML2_SSO_Redirect_Binding.jmx"
     [tenantMode]=false
     [skip]=false
     [modes]="FULL QUICK PUBLISH"
 )
-declare -A test_scenario4=(
-    [name]="04-oidc_auth_code_redirect_with_consent_retrieve_user_attributes"
+declare -A test_scenario7=(
+    [name]="07-Token_Exchange_Grant"
+    [display_name]="Token Exchange Grant"
+    [description]="Obtain an access token and an id token using the OAuth Token Exchange grant type."
+    [jmx]="oauth/Token_Exchange_Grant.jmx"
+    [tenantMode]=false
+    [skip]=false
+    [modes]="FULL QUICK PUBLISH"
+)
+
+# Test scenarions not publishing
+declare -A test_scenario8=(
```

Comment:
Shall we take this test also near to the without-consent tests.

Reply:
094b4b03db01bea11b7eb1c864483f6c32a3704d